app/api/get_apiURL.py

```diff
@@ -0,0 +1,136 @@
+import json
+from app.LLM_extractions.extractions import extract_information
+from app.api.validations import (
+    validate_age_order,
+    validate_diagnosis_and_control,
+)
+from app.fetch_termURLs.get_termURLs import (
+    get_diagnosis_termURL,
+    get_assessment_termURL,
+    get_sex_termURL,
+    get_image_modality_termURL,
+)
+
+
+def getApiURL(user_query: str) -> str:
+    """
+    Constructs the API URL using extracted parameters from the user's query.
+
+    Args:
+        user_query (str): The query provided by the user.
+
+    Returns:
+        str: The constructed API URL.
+    """
+    base_api_url = "https://api.neurobagel.org/query/?"
+    llm_response = extract_information(user_query)
+
+    try:
+        if isinstance(llm_response, str):
```

Comment:
Is this check in case the llm doesn't return a valid (string) response after 3 attempts?

Reply:
Yes, it handles that case.

Reply:
I would suggest modifying the error message to reflect the failure on the llm side maybe something like `I'm sorry, but I couldn't find the information you're looking for. Could you provide more details or clarify your question?` for a user from a conversational chatbot and if this is going to be an error message for the developer `llm failed to return a valid response`